### PR TITLE
Server db dev

### PR DESCRIPTION
**Pull request by lfelberg**
Adding changes recommended by Toby. More to come!

### CODE REVIEW

File: client/components/Player.jsx [hunk 1]
```diff
@@ -133,7 +102,8 @@ class Player extends React.Component {
   }
```
Comment (gkim795) [inline client/components/Player.jsx]: the handlebarhover and exit seem to have really similar lines. suggestion would be to combine it into one function that can handle both

File: client/components/SongBar.jsx [hunk 2]
```diff
@@ -12,24 +12,22 @@ const SongBar = ({
...
-        handleExit={(location === 'upper') ? handleExit : () => {}}
         handleClick={(location === 'upper') ? handleClick : () => {}}
         key={`bar_${location}_${i}`}
       />
       ));
     });
   });
   const handleBarClick = (e) => {
-    console.log('in bar', Object.keys(e.target.className, e.target.id), e.screenX, e.target.offsetLeft, e.target.offsetWidth);
+    console.log('in bar', e.target.offsetLeft, e.target.offsetWidth);
```
Comment (gkim795) [inline client/components/SongBar.jsx:30]: just a side note to remove console log when we finish our coding

File: client/components/SongBarEntry.jsx [hunk 3]
```diff
@@ -13,21 +13,19 @@ const SongBarEntry = ({
...
   position: PropTypes.string.isRequired,
-  bar: PropTypes.number.isRequired,
-  number: PropTypes.number.isRequired,
+  barHeight: PropTypes.number.isRequired,
   playState: PropTypes.object.isRequired,
   barFraction: PropTypes.number.isRequired,
   handleScan: PropTypes.func.isRequired,
-  handleExit: PropTypes.func.isRequired,
   handleClick: PropTypes.func.isRequired,
 };
```
Comment (gkim795) [inline client/components/SongBarEntry.jsx:31]: consider having default prop values - i used them as a way to pass some test if some data is not passed through

File: database/controllers.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    songs.tag,
+    songs.title,
+    sound_profiles.height,
+    sound_profiles.profile
+  `;
+  const queryString = `SELECT ${select} FROM songs INNER JOIN artists ON songs.id = ? `
+    + ' AND songs.artist_id = artists.id'
+    + ' INNER JOIN sound_profiles ON sound_profiles.song_id = songs.id';
+  return connection.query(queryString, songId);
+};
```
Comment (gkim795) [inline database/controllers.js:19]: are you doing '_' for database stuff and '.' for the react items? there seems to be slight inconsistency

File: server/index.js [hunk 5]
```diff
@@ -17,50 +13,7 @@ app.use((req, res, next) => {
...
-  db.getArtists(ids)
-    .then((artists) => {
-      res.status(200).send(JSON.stringify({ data: artists }));
-    })
-    .catch((err) => {
-      res.status(404).send(err);
-    });
-});
+// TODO: is this the best way to do this?
+app.use('/sc', songRoute);
```
Comment (gkim795) [inline server/index.js:17]: i love the routes option

File: server/routes.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
+const express = require('express');
+const db = require('../database/controllers');
+
+const router = express.Router();
+
+const queryAll = (req, res) => {
+  const { song } = req.params;
+  let requestEnd = req.originalUrl.split('/');
+  requestEnd = requestEnd[requestEnd.length - 1];
+  const dbQuery = (requestEnd === 'comments') ? db.getComments : db.getSong;
```
Comment (gkim795) [inline server/routes.js]: interesting method 
however, the requestEnd variable name might be unclear i would suggest to rename it to be clearer i.e.request or requestValue(?) etc

File: server/routes.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+  dbQuery(song)
+    .then((dbFind) => {
+      const returnData = (requestEnd === 'comments') ? dbFind : dbFind[0];
+      res.status(200).send(JSON.stringify({ data: returnData }));
```
Comment (gkim795) [inline server/routes.js:13]: dbFind[0] is the first index in the array returned by the query right? 
suggestion to make it a variable to be more clear

File: spec/components/songBarEntry.test.js [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+    component.find('div').simulate('click');
+    expect(props.handleClick).toHaveBeenCalled();
+  });
+
+  test('respond to pointerOver', () => {
+    const component = shallow(<SongBarEntry {...props} />);
+    component.find('div').simulate('pointerOver');
+    expect(props.handleScan).toHaveBeenCalled();
+  });
+});
```
Comment (gkim795) [inline spec/components/songBarEntry.test.js:69]: do the bars always go upper and then lower or is there ever a case with the same height?

File: spec/helpers/playerHelpers.test.js [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+    expect(helpers.convertToMinSec(241)).toBe('4:01');
+  });
+  test('add zeros when needed to minutes', () => {
+    expect(helpers.convertToMinSec(31)).toBe('0:31');
+  });
+  test('force non-numbers to zero', () => {
+    expect(helpers.convertToMinSec('hi')).toBe('0:00');
+  });
+});
+
```
Comment (gkim795) [inline spec/helpers/playerHelpers.test.js:48]: is there a case for if the time is longer than an hour?

File: client/components/Player.jsx [hunk 10]
```diff
@@ -3,26 +3,21 @@ import SongDisplay from './SongDisplay.jsx';
 import SongTracker from './SongTracker.jsx';
 import helpers from '../helpers/playerHelpers.js';
 
+const nSongs = 100;
```
Comment (gkim795) [inline client/components/Player.jsx]: nSongs is the count of the songs correct?

**Review by gkim795 — APPROVED** [review]
Main comments focused on variable names and slight suggestions to avoid repetition of code

File: client/components/SongBar.jsx (hunk 2 above)
Comment (lfelberg) [reply client/components/SongBar.jsx:30]: YES! This feature is still a little buggy, so I have them in there for now!

File: database/controllers.js (hunk 4 above)
Comment (lfelberg) [reply database/controllers.js:19]: Changed the names in a later update to all be CamelCase

File: client/components/Player.jsx (hunk 1 above)
Comment (lfelberg) [reply client/components/Player.jsx]: I like this! It's a good idea! I'm implementing it

File: server/routes.js (hunk 6 above)
Comment (lfelberg) [reply server/routes.js]: I talked to Toby and I am going to split these two back up for clarity

File: spec/components/songBarEntry.test.js (hunk 8 above)
Comment (lfelberg) [reply spec/components/songBarEntry.test.js:69]: There is an array of upper bars, and an array of lower bars, but the lower ones will be covered (partially) by the comments!

File: spec/helpers/playerHelpers.test.js (hunk 9 above)
Comment (lfelberg) [reply spec/helpers/playerHelpers.test.js:48]: There is not! I'll add that in.

File: client/components/Player.jsx (hunk 10 above)
Comment (lfelberg) [reply client/components/Player.jsx]: Yes, the count of songs in our test DB

File: server/routes.js (hunk 7 above)
Comment (lfelberg) [reply server/routes.js:13]: Changed it from dbFind to dbData

File: client/components/SongDisplay.jsx [hunk 11]
```diff
@@ -5,32 +5,37 @@ import SongArt from './SongArt.jsx';
...
 
 SongDisplay.propTypes = {
   song: PropTypes.object.isRequired,
   playing: PropTypes.bool.isRequired,
   handleAlbumClick: PropTypes.func.isRequired,
   handlePlayClick: PropTypes.func.isRequired,
-  handleInfoClick: PropTypes.func.isRequired,
 };
 
 export default SongDisplay;
```
Comment (gkim795) [inline client/components/SongDisplay.jsx:41]: new line
```suggestion
export default SongDisplay;

```

File: spec/components/songBarEntry.test.js (hunk 8 above)
Comment (gkim795) [reply spec/components/songBarEntry.test.js:69]: oohh i get what the upper/lower mean. sounds good!

**Review by gkim795 — APPROVED** [review]
There's some tests that are commented out but I haven't added any suggestions on that because I know you'll get rid of it or add on to it in the future. It's looking really good and I don't find your code hard to understand.

**Pull request merged**